lib/src/connection_checker.dart

```diff
@@ -83,14 +97,17 @@ class ConnectionChecker {
     Mapper<ConnectivityResult, ConnectionType>? connectionTypeMapper,
   })  : _checkHostReachability = checkHostReachability,
         _checkAddresses = checkAddresses,
+        _checkApiUrls = checkApiUrls,
         _checkOverDnsTimeout = checkOverDnsTimeout,
         _baseUrlLookupAddress = baseUrlLookupAddress,
         _failureAttempts = failureAttempts ?? _defaultFailureAttempts,
         _requestInterval = requestInterval ?? _defaultRequestInterval,
         _connectivity = connectivity ?? Connectivity(),
         _connectionTypeMapper = connectionTypeMapper ?? ConnectionTypeMapper(),
-        _hostReachabilityChecker =
-            hostReachabilityChecker ?? DefaultHostReachabilityChecker();
+        _hostReachabilityChecker = hostReachabilityChecker ??
+            (kIsWeb
+                ? WebHostReachabilityChecker()
+                : DefaultHostReachabilityChecker());
```

Comment:
[Suggestion] Check platform-specific imports based on the HTTP package. In my opinion, it is a more elegant solution than checking the isWeb constant. Stub: https://github.com/dart-lang/http/blob/master/pkgs/http/lib/src/client_stub.dart, and client impl that has conditional imports of platform specific functionality https://github.com/dart-lang/http/blob/master/pkgs/http/lib/src/client.dart